### ISSUE TITLE
Directives getting removed from the module schema

### ISSUE THREAD

**Issue by nsattler** [Urigo/graphql-modules#453]
Using directives and enums in modules lead to 
`GraphQL-Modules Error: Schema is not valid` with reason `Unknown directive "XXX"`

_The directives stopped working after upgrading from version **`0.6.5`**_

While the following is working (the base for the example is taken from the unit tests):

```typescript
import { SchemaDirectiveVisitor } from 'graphql-tools'
import { defaultFieldResolver, GraphQLString, execute } from 'graphql'
import { GraphQLModule } from '@graphql-modules/core'
import { gql } from 'apollo-server-core'

class FormattableDateDirective extends SchemaDirectiveVisitor {
  public visitFieldDefinition (field) {
    const { resolve = defaultFieldResolver } = field
    field.resolve = async function (source, args, context, info) {
      const date = await resolve.call(this, source, args, context, info)
      return date.toLocaleDateString()
    }
    field.type = GraphQLString
  }
}

const DateDirectiveModule = new GraphQLModule({
  name: 'DateDirectiveModule',
  typeDefs: gql`
    directive @date on FIELD_DEFINITION
  `,
  schemaDirectives: {
    date: FormattableDateDirective
  }
})

const VisitedDateModule = new GraphQLModule({
  name: 'VisitedDateModule',
  typeDefs: gql`
    scalar Date
    type Query {
      today: Date @date
    }
  `,
  resolvers: {
    Query: {
      today: () => new Date()
    }
  },
  imports: [DateDirectiveModule]
})

const { schema, schemaDirectives } = new GraphQLModule({
  name: 'appModule',
  imports: [VisitedDateModule]
})

SchemaDirectiveVisitor.visitSchemaDirectives(schema, schemaDirectives)

const main = async () => {
  const result = await execute({
    schema,
    document: gql`
      query {
        today
      }
    `
  })
  console.log(result)
}

main().then(() => console.log('END')).catch((err) => console.log(err))
```

As soon as I introduce an enum it stops working:

```typescript
import { SchemaDirectiveVisitor } from 'graphql-tools'
import { defaultFieldResolver, GraphQLString, execute } from 'graphql'
import { GraphQLModule } from '@graphql-modules/core'
import { gql } from 'apollo-server-core'

class FormattableDateDirective extends SchemaDirectiveVisitor {
  public visitFieldDefinition (field) {
    const { resolve = defaultFieldResolver } = field
    const { format } = this.args
    field.resolve = async function (source, args, context, info) {
      const date = await resolve.call(this, source, args, context, info)
      if (format === 'local') {
        return date.toLocaleDateString()
      }
      return date.toISOString()
    }

    field.type = GraphQLString
  }
}

const DateDirectiveModule = new GraphQLModule({
  name: 'DateDirectiveModule',
  typeDefs: gql`
    directive @date(format: DateFormat) on FIELD_DEFINITION
    enum DateFormat {
      LOCAL
      ISO
    }
  `,
  schemaDirectives: {
    date: FormattableDateDirective
  },
  resolvers: {
    DateFormat: {
      LOCAL: 'local',
      ISO: 'iso'
    }
  }
})

const VisitedDateModule = new GraphQLModule({
  name: 'VisitedDateModule',
  typeDefs: gql`
    scalar Date
    type Query {
      today: Date @date
    }
  `,
  resolvers: {
    Query: {
      today: () => new Date()
    }
  },
  imports: [DateDirectiveModule]
})

const { schema, schemaDirectives } = new GraphQLModule({
  name: 'appModule',
  imports: [VisitedDateModule]
})

SchemaDirectiveVisitor.visitSchemaDirectives(schema, schemaDirectives)

const main = async () => {
  const result = await execute({
    schema,
    document: gql`
      query {
        today
      }
    `
  })
  console.log(result)
}

main().then(() => console.log('END')).catch((err) => console.log(err))
```

The output is then

```
Error: 
      GraphQL-Modules Error: Schema is not valid!
      - #Module #VisitedDateModule doesn't have a valid schema!
      -- Unknown directive "date".

      Possible solutions:
      - Check syntax errors in typeDefs
      - Make sure you import correct dependencies
    
    at GraphQLModule.get schema [as schema] (/.../node_modules/@graphql-modules/core/src/graphql-module.ts:306:17)
    at selfImports.map.module (/.../node_modules/@graphql-modules/core/src/graphql-module.ts:273:63)
    at Array.map (<anonymous>)
    at GraphQLModule.get schema [as schema] (/.../node_modules/@graphql-modules/core/src/graphql-module.ts:273:42)
    ...
```

A little bit debugging showed me that basically the directives get removed from schema as soon as resolvers for enum are applied to the schema.
Regardless of having them explicitly defined in resolvers or using the default ones.

This happens in the transformation graphql-tools/transforms/ConvertEnumValues which is always applied to the schema if enums are used.

I've found a small trick on fixing this abnormal behaviour using the middleware option and re-attaching directives back to the schema,
but this feels rather like a hack than a solution :)

Here we go:

```typescript
import { SchemaDirectiveVisitor } from 'graphql-tools'
import { defaultFieldResolver, GraphQLString, execute, visit, extendSchema } from 'graphql'
import { GraphQLModule } from '@graphql-modules/core'
import { gql } from 'apollo-server-core'

const directiveFixingMiddleware = ({ schema, typeDefs }) => {
  const directiveTypeDefs = visit(typeDefs, {
    enter: (node) => {
      if (node.kind === 'Document') return
      if (node.kind === 'DirectiveDefinition') {
        const foundDirective = schema.getDirective(node.name.value)
        return foundDirective ? null : false
      }
      return null
    }
  })
  if (directiveTypeDefs.definitions && directiveTypeDefs.definitions.length > 0) {
    const fixedSchema = extendSchema(schema, directiveTypeDefs)
    return { schema: fixedSchema }
  }
  return { schema }
}

class FormattableDateDirective extends SchemaDirectiveVisitor {
  public visitFieldDefinition (field) {
    const { resolve = defaultFieldResolver } = field
    const { format } = this.args
    field.resolve = async function (source, args, context, info) {
      const date = await resolve.call(this, source, args, context, info)
      if (format === 'local') {
        return date.toLocaleDateString()
      }
      return date.toISOString()
    }

    field.type = GraphQLString
  }
}

const DateDirectiveModule = new GraphQLModule({
  name: 'DateDirectiveModule',
  middleware: directiveFixingMiddleware,
  typeDefs: gql`
    directive @date(format: DateFormat) on FIELD_DEFINITION
    enum DateFormat {
      LOCAL
      ISO
    }
  `,
  schemaDirectives: {
    date: FormattableDateDirective
  },
  resolvers: {
    DateFormat: {
      LOCAL: 'local',
      ISO: 'iso'
    }
  }
})

const VisitedDateModule = new GraphQLModule({
  name: 'VisitedDateModule',
  middleware: directiveFixingMiddleware,
  typeDefs: gql`
    scalar Date
    type Query {
      today: Date @date(format: LOCAL)
    }
  `,
  resolvers: {
    Query: {
      today: () => new Date()
    }
  },
  imports: [DateDirectiveModule]
})

const { schema, schemaDirectives } = new GraphQLModule({
  name: 'appModule',
  middleware: directiveFixingMiddleware,
  imports: [VisitedDateModule]
})

SchemaDirectiveVisitor.visitSchemaDirectives(schema, schemaDirectives)

const main = async () => {
  const result = await execute({
    schema,
    document: gql`
      query {
        today
      }
    `
  })
  console.log(result)
}

main().then(() => console.log('END')).catch((err) => console.log(err))
```

**Comment by kamilkisiela** [comment]
Test https://github.com/Urigo/graphql-modules/pull/497

**Comment by ardatan** [comment]
Fixed in 0.7.6!
